typed-racket-lib/typed-racket/utils/redirect-contract.rkt

```diff
@@ -0,0 +1,45 @@
+#lang racket/base
+
+(require syntax/modcollapse (for-template racket/base))
+(provide make-make-redirect-to-contract)
+
+;; This produces an identifier that expands to a local-require.  It's
+;; used to implement identifiers that are protected on export from TR
+;; with contracts, but where the contracted defintion is in the
+;; #%contract-defs submodule.
+
+;; varref: a variable reference to the typed module that has the
+;;         appropriate submodule in it
+
+;; id: the name of the export from the submodule which will be
+;;     redirected-to by the local-require
+
+;; stx: the syntax object that's the argument to the macro (that is,
+;;      the stx object that's the reference to the typed identifier in
+;;      a untyped module). The funny eta-expansion with `redirect` is
```

Comment:
I don't see why you need the eta expansion, just remove the `stx` argument and return `redirect`. This also makes it clearer that this is returning a syntax transformer.